packages/dev/core/src/Cameras/flyCamera.ts

```diff
@@ -339,16 +341,24 @@ export class FlyCamera extends TargetCamera {
         this._collider._radius = this.ellipsoid;
         this._collider.collisionMask = this._collisionMask;
 
-        // No need for clone, as long as gravity is not on.
-        let actualDisplacement = displacement;
+        // Copy displacement vector into temporary to manipulate it
+        const scaledDisplacement = TmpVectors.Vector3[0].copyFrom(displacement);
 
-        // Add gravity to direction to prevent dual-collision checking.
+        //add gravity to the direction to prevent the dual-collision checking
         if (this.applyGravity) {
-            // This prevents mending with cameraDirection, a global variable of the fly camera class.
-            actualDisplacement = displacement.add(this.getScene().gravity);
+            // Apply gravity to current displacement
+            scaledDisplacement.addInPlace(this.getScene().gravity);
         }
 
-        coordinator.getNewPosition(this._oldPosition, actualDisplacement, this._collider, 3, null, this._onCollisionPositionChange, this.uniqueId);
+        // Scale original displacement by relative inertia (decay over time to zero)
+        displacement.scaleInPlace(relativeInertia);

```

Comment:
I think this is a breaking change? The old code did not update `displacement`.

Reply:
Since this change is a bit weird, I've modified the code to hopefully be more readable

Reply:
Just to elaborate more on this, `displacement` is no longer being modified in this function